src/Common/AzurePSCmdlet.cs

```diff
@@ -365,6 +366,32 @@ protected override void BeginProcessing()
             //Now see if the cmdlet has any Breaking change attributes on it and process them if it does
             //This will print any breaking change attribute messages that are applied to the cmdlet
             BreakingChangeAttributeHelper.ProcessCustomAttributesAtRuntime(this.GetType(), this.MyInvocation, WriteWarning);
+
+            DisplayAzureRmRetiringMessage();
+        }
+
+        /// <summary>
+        /// Display warning message that AzureRM will retire unless customers set environment variable 'SuppressAzureRmModulesRetiringWarning' as true.
+        /// </summary>
+        private void DisplayAzureRmRetiringMessage()
+        {
+            bool supressRetireWarningMessage = false;
+
+            try
+            {
+                supressRetireWarningMessage = bool.Parse(Environment.GetEnvironmentVariable("SuppressAzureRmModulesRetiringWarning"));
+            }
+            catch (Exception)
+            {
+                //no action
+            }
```

Comment:
How about using tryParse()
bool.TryParse(Environment.GetEnvironmentVariable("SuppressAzureRmModulesRetiringWarning"), out supressRetireWarningMessage);

Reply:
Good suggestion. However Environment.GetEnvironmentVariable("SuppressAzureRmModulesRetiringWarning") may throw exception per https://docs.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariable?view=net-5.0, we need to use try catch anyway...

Reply:
Make sense. Thank your for clarification.